src/types/web-resource.ts

```diff
@@ -0,0 +1,140 @@
+import * as TypeUtils from '../type-utils';
+
+export type WebResource = {
+	filename: string;
+	href: string;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
+};
+
+export const types = {
+	postgres: 'JSONB',
+	mysql: 'JSON',
+	websql: 'TEXT',
+	odata: {
+		name: 'Self.WebResource',
+		complexType: `\
+<ComplexType Name="WebResource">
+	<Property Name="filename" Nullable="false" Type="Edm.String"/>\
+	<Property Name="href" Nullable="false" Type="Edm.String"/>\
+	<Property Name="contentType" Nullable="true" Type="Edm.String"/>\
+	<Property Name="contentDisposition" Nullable="true" Type="Edm.String"/>\
+	<Property Name="size" Nullable="true" Type="Edm.Int64"/>\
+</ComplexType>`,
+	},
+};
+
+export const nativeProperties = {
+	has: {
+		Filename: (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'filename']],
+		],
+		HRef: (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'href']],
+		],
+		'Content Type': (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'contentType']],
+		],
+		'Content Disposition': (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'contentDisposition']],
+		],
+		Size: (stringifiedResource: string) => [
+			'Cast',
+			[
+				'ExtractJSONPathAsText',
+				stringifiedResource,
+				['TextArray', ['EmbeddedText', 'size']],
+			],
+			'Integer',
+		],
+	},
+};
+
+/**
+ * Converts the data, which comes from the DB as a string or object depending on the
+ * column type, to a WebResource object
+ *
+ * @param data string|object
+ * @returns a WebResource parsed from the DB
+ */
+export const fetchProcessing = (data: any) => {
+	let refData: WebResource;
+	if (data === null) {
+		return data;
+	}
+	if (typeof data === 'string') {
+		try {
+			refData = JSON.parse(data);
+		} catch (e: any) {
+			throw new Error(
+				`can't be parsed from stored value ${typeof data} with error ${
+					e.message
```

Comment:
Another potential error message leak

Reply:
fixed

Reply:
But I wonder if this actually throws... maybe only if you store as plain text?